willa/etl/pipeline.py

```diff
@@ -0,0 +1,28 @@
+"""
+Run the Willa ETL pipeline.
+"""
+
+from langchain_core.vectorstores import InMemoryVectorStore
+from langchain_core.vectorstores.base import VectorStore
+from langchain_ollama import OllamaEmbeddings
+
+from .doc_proc import load_pdfs, split_all_docs, embed_docs
+
+
+def run_pipeline(vector_store: VectorStore | None = None) -> VectorStore:
+    """Run the ETL pipeline for Willa.
+
+    :param vector_store: The vector store in which to store processed documents.
+                         If no vector store is specified, a new
+                         in-memory vector store will be created.
+    :returns: The vector store where processed documents are stored.
+    """
+    if vector_store is None:
+        embeddings = OllamaEmbeddings(model='nomic-embed-text')
```

Comment:
similar question as to `willa.chatbot.Chatbot` - can we make this more generic to accept a parameter that will accept a subclass of `langchain_core.embeddings.embeddings.Embeddings`?

Reply:
For this one, this is a fallback if no vector store is specified - which is assumed to be a local development instance.  I'm not sure it makes sense to parameterise this since if you want a different embedding you can pass your own vector store in (including an InMemoryVectorStore that uses a different embedding model).

If we want to drop hardcoding Ollama entirely - as in, have absolutely no mention of it in the code base - we can do that, but I'm not sure how the CLI will work then.  I was trying to make it as "low touch" as possible for both dev and prod.